cmd/esc/cli/cli_test.go

```diff
@@ -990,8 +992,22 @@ func (c *testExec) runScript(script string, cmd *exec.Cmd) error {
 				esc.SetIn(hc.Stdin)
 				esc.SetOut(hc.Stdout)
 				esc.SetErr(hc.Stderr)
-				if err := esc.Execute(); err != nil {
+
+				ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+				esc.SetContext(ctx)
+				defer cancel()
+
+				ch := make(chan error, 1)
+				go func() {
+					ch <- esc.Execute()
+				}()
+				select {
+				case <-ctx.Done():
 					return interp.NewExitStatus(1)
```

Comment:
Can't add suggestions since this line didn't change, but we should return the context's error here--it'll be a bit more informative:

```golang
					return ctx.Err()
```